pages/posts/[filename].tsx

```diff
@@ -35,17 +37,19 @@ export const query = (gql) => gql`
   }
 `;
 
-export type PostQueryResponseType = {
-  getPostsDocument: Posts_Document;
-};
-
 const client = createLocalClient();
 
-export const getStaticProps = async ({ params }) => {
+export const getStaticProps = async ({ params, preview }) => {
+  const variables = { relativePath: `${params.filename}.md` };
   return {
-    props: await client.request(query, {
-      variables: { relativePath: `${params.filename}.md` },
-    }),
+    props: {
+      data: await client.request<{ getPostsDocument: Posts_Document }>(query, {
+        variables,
+      }),
+      variables,
+      query,
+      preview: !!preview,
+    },
```

Comment:
This shape is required for _all_ `getStaticProps` return values. Perhaps we could be a helper function or typescript type